From 5cdb216a16e84c9b1a85283bc389f8e683bfd1af Mon Sep 17 00:00:00 2001
From: FrancescoFerraro <francesco.f@variscite.com>
Date: Wed, 28 Feb 2024 09:42:28 +0100
Subject: [PATCH 1/3] android-imx_device_fsl: allow building other NXP
 derivatives

These changes assume that the NXP derivatives are at the same level of
NXP device. For Variscite SoMs this allows reusing most of the files
provided by NXP and just change what really required.

Derived from:
  https://github.com/varigit/MX6x-android/blob/android-12.0.0_1.0.0-var02/scripts/platform/device/nxp.git/0001-android-imx_device_fsl-allow-building-other-NXP-deri.patch

Signed-off-by: FrancescoFerraro <francesco.f@variscite.com>
---
 common/build/Makefile    |  4 ++--
 common/tools/imx-make.sh | 10 +++++-----
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/common/build/Makefile b/common/build/Makefile
index ae046dbc..5d58adf4 100644
--- a/common/build/Makefile
+++ b/common/build/Makefile
@@ -15,8 +15,8 @@ include ${soc_path}/UbootKernelCommonConfig.mk
 include ${product_path}/SharedBoardConfig.mk
 include ${product_path}/UbootKernelBoardConfig.mk
 
-include ${nxp_git_path}/common/build/kernel.mk
-include ${nxp_git_path}/common/build/uboot.mk
+include ${nxp_git_path}/../nxp/common/build/kernel.mk
+include ${nxp_git_path}/../nxp/common/build/uboot.mk
 -include ${product_path}/AndroidTee.mk
 -include ${GPU_VIV6_PATH}/gpu-viv6/driver/hal/kernel/galcore.mk
 -include ${VVCAM_PATH}/vvcam/vvcam.mk
diff --git a/common/tools/imx-make.sh b/common/tools/imx-make.sh
index c2cb5654..dd795bcd 100755
--- a/common/tools/imx-make.sh
+++ b/common/tools/imx-make.sh
@@ -141,7 +141,7 @@ if [ -n "${build_kernel_modules}" ]; then
     build_kernel_oot_module_flag=1;
 fi
 
-product_makefile=`pwd`/`find device/nxp -maxdepth 4 -name "${TARGET_PRODUCT}.mk"`;
+product_makefile=`pwd`/`find device -maxdepth 4 -name "${TARGET_PRODUCT}.mk"`;
 product_path=${product_makefile%/*}
 soc_path=${product_path%/*}
 nxp_git_path=${soc_path%/*}
@@ -158,7 +158,7 @@ fi
 
 # redirect standard input to /dev/null to avoid manually input in kernel configuration stage
 soc_path=${soc_path} product_path=${product_path} nxp_git_path=${nxp_git_path} clean_build=${clean_build} \
-    make -C ./ -f ${nxp_git_path}/common/build/Makefile ${parallel_option} \
+    make -C ./ -f ${nxp_git_path}/../nxp/common/build/Makefile ${parallel_option} \
     ${build_bootloader} ${build_kernel} </dev/null || exit
 # in the execution of this script, if the kernel build env is cleaned or configured, do not trigger that again
 if [ -n "${build_kernel}" ]; then
@@ -168,21 +168,21 @@ fi
 
 if [ -n "${build_kernel_modules}" ]; then
     soc_path=${soc_path} product_path=${product_path} nxp_git_path=${nxp_git_path} clean_build=${clean_build} \
-        skip_config_or_clean=${skip_config_or_clean} make -C ./ -f ${nxp_git_path}/common/build/Makefile ${parallel_option} \
+        skip_config_or_clean=${skip_config_or_clean} make -C ./ -f ${nxp_git_path}/../nxp/common/build/Makefile ${parallel_option} \
         ${build_kernel_modules} </dev/null || exit
     skip_config_or_clean=1
 fi
 
 if [ -n "${build_kernel_dts}" ]; then
     soc_path=${soc_path} product_path=${product_path} nxp_git_path=${nxp_git_path} clean_build=${clean_build} \
-        skip_config_or_clean=${skip_config_or_clean} make -C ./ -f ${nxp_git_path}/common/build/Makefile ${parallel_option} \
+        skip_config_or_clean=${skip_config_or_clean} make -C ./ -f ${nxp_git_path}/../nxp/common/build/Makefile ${parallel_option} \
         ${build_kernel_dts} </dev/null || exit
     skip_config_or_clean=1
 fi
 
 if [ ${build_kernel_oot_module_flag} -eq 1 ] || [ -n "${build_kernel_modules}" ]; then
     soc_path=${soc_path} product_path=${product_path} nxp_git_path=${nxp_git_path} clean_build=${clean_build} \
-        skip_config_or_clean=${skip_config_or_clean} make -C ./ -f ${nxp_git_path}/common/build/Makefile ${parallel_option} \
+        skip_config_or_clean=${skip_config_or_clean} make -C ./ -f ${nxp_git_path}/../nxp/common/build/Makefile ${parallel_option} \
         ${build_vvcam} ${build_galcore} ${build_mxmwifi} </dev/null || exit
 fi
 
-- 
2.43.2

